catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroMyProjectsActivityTest.java

```diff
@@ -0,0 +1,153 @@
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.NoteBrick;
+import org.catrobat.catroid.io.StorageHandler;
+import org.catrobat.catroid.phiro.ui.PhiroMainMenuActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.uiespresso.ui.activity.utils.ProjectListDataInteractionWrapper.onProjectWithName;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroMyProjectsActivityTest {
+
+	private IdlingResource idlingResource;
+
+	@Rule
+	public BaseActivityInstrumentationRule<PhiroMainMenuActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(PhiroMainMenuActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		deletePhiroPrograms();
+		baseActivityTestRule.launchActivity(null);
+
+		idlingResource = baseActivityTestRule.getActivity().getIdlingResource();
+		Espresso.registerIdlingResources(idlingResource);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		Espresso.unregisterIdlingResources(idlingResource);
+	}
+
+	@Test
+	public void phiroProjectsInitiallyLoadedTest() {
+		openProjectList();
+
+		for (String projectName : PhiroMainMenuActivity.phiroProjects) {
+			onProjectWithName(projectName)
+					.check(matches(isDisplayed()));
+		}
+	}
+
+	@Test
+	public void phiroProgramRestoredWhenDeletedTest() {
+		String projectName = PhiroMainMenuActivity.phiroProjects[0];
+
+		openProjectList();
+
+		StorageHandler.getInstance().deleteProject(projectName);
+
+		onView(withId(android.R.id.home))
+				.perform(click());
+
+		openProjectList();
+
+		onProjectWithName(projectName)
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void phiroProgramsUnchangedWhenModifiedTest() throws InterruptedException {
+		openProjectList();
+
+		addNoteBrickToFirstScript(PhiroMainMenuActivity.phiroProjects[0]);
+
+		onView(withId(android.R.id.home))
+				.perform(click());
+
+		openProjectList();
+
+		checkAddedNoteBrickIsStillInFirstScript();
+	}
+
+	private void openProjectList() {
```

Comment:
since all tests start with a call to "openProjectList", refactor the test to open the fragment or activity you actually want to test instead of the main menu activity and delete this function.